src/core/qgsofflineediting.cpp

```diff
@@ -1096,6 +1097,14 @@ void QgsOfflineEditing::updateFidLookup( QgsVectorLayer *remoteLayer, sqlite3 *d
 
 void QgsOfflineEditing::copySymbology( QgsVectorLayer *sourceLayer, QgsVectorLayer *targetLayer )
 {
+  const QMap<QString, QgsMapLayerStyle> styles = sourceLayer->styleManager()->mapLayerStyles();
```

Comment:
Would it be cleaner for all this code to be moved to QgsVectorLayer? Eg if it was renamed to copyStyleFromLayer and accepted the style categories enum flag, I think there's likely existing code in qgsvectorlayer which could be reused for the guts of this function....

Reply:
Moved to QgsMapLayerStyleManager.
Adding the style categories was not straightforward, as this class heavily depends on (de-)serialized styles and all the handling of this flag is done on a different level. To do this cleanly it would require a quite profound refactoring of how styles are handled or copying a lot of code from Qgs*Layer to here, both of which are clearly out of scope for this fix.

Reply:
What you've done looks good!

I was actually meaning something a bit more generic, and moving copySymbology across to QgsVectorLayer (or QgsMapLayer)... but it's up to you

Reply:
I'm not sure I understand where you are going. Is it just moving the method to QgsMapLayer directly? That shouldn't make any difference IMO?
I could also see benefit for something like this in the UI, but this would be new functionality and could also use the current implementation.

Reply:
It wasn't about exposing this anywhere in UI -- just about API convenience. If I was looking for an existing API method to copy symbology settings from one layer to another, I definitely wouldn't think to look in the QgsOfflineEditing class :laughing: !

Reply:
It's in map layer style manager since your proposal.